packages/app-mobile/components/NoteEditor/ImageEditor/ImageEditor.tsx

```diff
@@ -0,0 +1,205 @@
+			}
+
+			body, html {
+				padding: 0;
+				margin: 0;
+			}
+		`;
+	}, [editorTheme]);
+};
+
+const ImageEditor = (props: Props) => {
+	const editorTheme: Theme = themeStyle(props.themeId);
+	const webviewRef = useRef(null);
+
+	const onRequestCloseEditor = useCallback(() => {
+		Alert.alert(
+			_('Save changes?'), _('This drawing may have unsaved changes.'), [
+				{
+					text: _('Discard changes'),
+					onPress: async () => {
+						await clearAutosave();
+						props.onCancel();
+					},
+					style: 'destructive',
+				},
+				{
+					text: _('Save changes'),
+					onPress: () => {
+						// saveDrawing calls props.onSave(...) which may close the
+						// editor.
+						webviewRef.current.injectJS('saveDrawing();');
+					},
+				},
+			]
+		);
+		return true;
+	}, [webviewRef, props.onCancel]);
+
+	useEffect(() => {
+		BackHandler.addEventListener('hardwareBackPress', onRequestCloseEditor);
+
+		return () => {
+			BackHandler.removeEventListener('hardwareBackPress', onRequestCloseEditor);
+		};
+	}, [onRequestCloseEditor]);
+
+	const css = useCss(editorTheme);
+	const html = useMemo(() => `
+		<!DOCTYPE html>
+		<html>
+			<head>
+				<meta charset="utf-8"/>
+				<meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1,user-scalable=no"/>
+
+				<style>
+					${css}
+				</style>
+			</head>
+			<body></body>
+		</html>
+	`, [css]);
+
+	const injectedJavaScript = useMemo(() => `
+		window.onerror = (message, source, lineno) => {
+			window.ReactNativeWebView.postMessage(
+				"error: " + message + " in file://" + source + ", line " + lineno
+			);
+		};
+
+		const saveDrawing = (isAutosave) => {
+			const img = window.editor.toSVG();
+			window.ReactNativeWebView.postMessage(
+				JSON.stringify({
+					action: isAutosave ? 'autosave' : 'save',
+					data: img.outerHTML,
+				}),
+			);
+		};
+		window.saveDrawing = saveDrawing;
+
+		const closeDrawing = () => {
+			window.ReactNativeWebView.postMessage('{ "action": "close" }');
+		};
+
+		try {
+			if (window.editor === undefined) {
+				${shim.injectedJs('svgEditorBundle')}
+
+				window.editor = svgEditorBundle.createJsDrawEditor();
+
+				window.initialSVGData = ${JSON.stringify(props.initialSVGData)};
+				if (initialSVGData && initialSVGData.length > 0) {
+					editor.loadFromSVG(initialSVGData);
+				}
+
+				const toolbar = editor.addToolbar();
+
+				toolbar.addActionButton({
+					label: ${JSON.stringify(_('Close'))}, 
+					icon: svgEditorBundle.makeCloseIcon(),
+				}, () => closeDrawing());
+
+				toolbar.addActionButton({
+					label: ${JSON.stringify(_('Done'))},
+					icon: editor.icons.makeSaveIcon(),
+				}, () => {
+					saveDrawing(false);
+				});
+
+				svgEditorBundle.restoreToolbarState(
+					toolbar,
+					${JSON.stringify(Setting.value('imageeditor.jsdrawToolbar'))}
+				);
+				svgEditorBundle.listenToolbarState(editor, toolbar);
+
+				// Auto-save every two minutes.
+				const autoSaveInterval = 2 * 60 * 1000;
```

Comment:
Is it necessary to auto-save? Especially every 2 minutes doesn't seem often enough to be useful? Or maybe it should be more frequent (or not done?)

Reply:
I think auto-saving every two minutes is reasonable. In large images, saving be slow-ish (e.g. 500 ms), so we don't want to do it too often.

I personally have been using `js-draw` to take two-column notes for some of my classes. Because these notes are all in one image, having auto-save is useful.
<details><summary>Example</summary>

![e893808e91a64c5d99b81cbe2f10ad80](https://user-images.githubusercontent.com/46334387/211581109-f917b35a-a5f0-4d27-bae7-acb3b77eb7a2.svg)

</details>

I could, of course, work on making saving faster, cumulative, or asynchronous.

Reply:
> I think auto-saving every two minutes is reasonable. In large images, saving be slow-ish (e.g. 500 ms), so we don't want to do it too often.

For this kind of usage 2 minutes indeed makes sense so for now let's keep it. But indeed what if it takes a long time to save, let's say more than 120 seconds - what would happen in this case? Should there be some guards to prevent it?

Maybe for now a simple "isSaving" flag and not started to save if it's "true" would be enough.